precompiles/referenda/Referenda.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.8.3;
+
+/// @dev The Referenda contract's address.
+address constant REFERENDA_ADDRESS = 0x0000000000000000000000000000000000000811;
+
+/// @dev The Referenda contract's instance.
+Referenda constant REFERENDA_CONTRACT = Referenda(REFERENDA_ADDRESS);
+
+/// @author The Moonbeam Team
+/// @title Pallet Referenda Interface
+/// @title The interface through which solidity contracts will interact with the Referenda pallet
+/// @custom:address 0x0000000000000000000000000000000000000811
+interface Referenda {
+    /// @dev Defines the referenda origins that have tracks corresponding to uint8 representation
+    /// The uint8 representation is defined in origins pallet Into<u16> for Origin
+    /// From top to bottom: 1, 10, 11, 12, 13, 14, 15
```

Comment:
In Solidity, we can't give a custom index to each variant, so we have to parse the different variants in the natural int order 0, 1, 2, 3, etc